myokit/formats/sbml/_importer.py

```diff
@@ -159,42 +188,101 @@ def model(self, path):
+            # MathML
+            ns['mathml'] = "{http://www.w3.org/1998/Math/MathML}"
+        else:
+            # SBML
+            ns['sbml'] = sbml_version
+            # MathML
+            ns['mathml'] = "{http://www.w3.org/1998/Math/MathML}"
+
+            # log import warning
+            log.warn('The SBML version %s has not been tested. The model'
+                     % sbml_version + ' may not be imported correctly.'
+                     )
+
+        return ns
+
+    def _get_sbml_version(self, root):
+        """
+        Returns the SBML version of the file.
+        """
+        m = re.match(r'\{.*\}', root.tag)
+
+        return m.group(0) if m else ''
+
+    def _parse_initial_assignments(self, model, comp, refs, node, ns):
         """
         Parses any initial values specified outside of the rules section.
         """
-        node = dom_child(node, 'initialAssignment')
-        while node:
-            var = str(node.getAttribute('symbol')).strip()
+        nodes = node.findall(ns + 'initialAssignment')
+        # get mathml ns
+        mathml_ns = self.ns['mathml']
+        # iterate through initial assignments
+        for node in nodes:
+            var = str(node.get('symbol')).strip()
             var = self._convert_name(var)
             if var in comp:
                 self.logger().log(
                     'Parsing initial assignment for "' + var + '".')
                 var = comp[var]
-                expr = parse_mathml_dom(
-                    dom_child(node, 'math'), refs, self.logger())
-
-                if var.is_state():
-                    # Initial value
-                    var.set_state_value(expr)
-                else:
-                    # Change of value
-                    var.set_rhs(expr)
+                # get child
+                child = node.find(mathml_ns + 'math')
+                if child:
+                    expr = parse_mathml_etree(
+                        child,
+                        lambda x, y: myokit.Name(refs[x]),
+                        lambda x, y: myokit.Number(x)
+                    )
+
+                    if var.is_state():
+                        # Initial value
+                        var.set_state_value(expr)
+                    else:
+                        # Change of value
+                        var.set_rhs(expr)
             else:   # pragma: no cover
                 raise SBMLError(   # pragma: no cover
                     'Initial assignment found for unknown parameter <' + var
                     + '>.')
 
-            node = dom_next(node, 'initialAssignment')
-
-    def _parse_parameters(self, model, comp, refs, node):
+    def _parse_parameters(self, model, comp, refs, node, ns):
         """
         Parses parameters
         """
-        node = dom_child(node, 'parameter')
-        while node:
+        nodes = node.findall(ns + 'parameter')
+        for node in nodes:
```

Comment:
Since `nodes` is never used again, you could also just write `for node in node.findall(..)

Reply:
Agreed, and changed everywhere.